tests/equilibria/test_physics.py

```diff
@@ -28,3 +32,30 @@ def test_psi_bd(self):
 
         assert round(abs(psib - 143), 0) == 0  # 142.66 ~ 143 V.s
         # CREATE DEMO 2015 test case
+
+
+class TestCoulombLogarithm:
+    @classmethod
+    def setup_class(cls):
+        df = pd.DataFrame(
+            columns=["Case", "n [1/m^3]", "T [eV]", "ln Lambda (Goldston)"]
+        )
+
+        df.loc[0] = ["Solar wind", 10.0**7, 10.0, 26]
+        df.loc[1] = ["Van Allen belts", 10.0**9, 10.0**2, 26]
+        df.loc[2] = ["Earth's ionosphere", 10.0**11, 10.0**-1, 14]
+        df.loc[3] = ["Solar corona", 10.0**13, 10.0**2, 21]
+        df.loc[4] = ["Gas discharge", 10.0**16, 10.0**0, 12]
+        df.loc[5] = ["Process plasma", 10.0**18, 10.0**2, 15]
+        df.loc[6] = ["Fusion experiment", 10.0**19, 10.0**3, 17]
+        df.loc[7] = ["Fusion reactor", 10.0**20, 10.0**4, 18]
+        df.loc[8] = ["Hartmut's case", 10.0**20, 1000.0, 16.5]
+        cls.data = df
+
+    def test_coulomb_logarithm_values(self):
+        for i in range(9):
+            temp_in_k = self.data.loc[i, "T [eV]"] * EV_TO_J / K_BOLTZMANN
+            density = self.data.loc[i, "n [1/m^3]"]
+            value = round(coulomb_logarithm(temp_in_k, density), 1)
+            reference_value = self.data.loc[i, "ln Lambda (Goldston)"]
+            np.testing.assert_allclose(value, reference_value, rtol=0.054)
```

Comment:
I would avoid asserting within a loop when possible, it can be a pain when a case fails, but you don't know which case it was. Also, is the `DataFrame` necessary here? Would it be simpler for the cases to be in a list? Suggestion below:

```python
     CASES = {
        # Case: [ n [1/m^3], T [eV], ln Lambda (Goldston) ]
        "Solar wind": [10.0**7, 10.0, 26],
        "Van Allen belts": [10.0**9, 10.0**2, 26],
        "Earth's ionosphere": [10.0**11, 10.0**-1, 14],
        "Solar corona": [10.0**13, 10.0**2, 21],
        "Gas discharge": [10.0**16, 10.0**0, 12],
        "Process plasma": [10.0**18, 10.0**2, 15],
        "Fusion experiment": [10.0**19, 10.0**3, 17],
        "Fusion reactor": [10.0**20, 10.0**4, 18],
        "Hartmut's case": [10.0**20, 1000.0, 16.5],
    }

    @pytest.mark.parametrize("case", CASES.values(), ids=CASES.keys())
    def test_coulomb_logarithm_values(self, case):
        density, temp, reference_value = case
        temp_in_k = temp * EV_TO_J / K_BOLTZMANN

        value = round(coulomb_logarithm(temp_in_k, density), 1)

        np.testing.assert_allclose(value, reference_value, rtol=0.054)
```